src/main/java/de/tum/in/www1/artemis/service/metis/conversation/auth/ChannelAuthorizationService.java

```diff
@@ -0,0 +1,265 @@
+import de.tum.in.www1.artemis.repository.metis.conversation.ChannelRepository;
+import de.tum.in.www1.artemis.security.Role;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.web.rest.errors.AccessForbiddenException;
+
+@Service
+public class ChannelAuthorizationService extends ConversationAuthorizationService {
+
+    private final ChannelRepository channelRepository;
+
+    public ChannelAuthorizationService(UserRepository userRepository, AuthorizationCheckService authorizationCheckService,
+            ConversationParticipantRepository conversationParticipantRepository, ChannelRepository channelRepository) {
+        super(conversationParticipantRepository, userRepository, authorizationCheckService);
+        this.channelRepository = channelRepository;
+    }
+
+    /**
+     * Checks if a user is allowed to create a channel in a course or throws an exception if not
+     *
+     * @param course the course the channel should be created in
+     * @param user   the user that wants to create the channel
+     */
+    public void isAllowedToCreateChannel(@NotNull Course course, @NotNull User user) {
+        var userToCheck = getUserIfNecessary(user);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.TEACHING_ASSISTANT, course, userToCheck);
+    }
+
+    /**
+     * Checks if a user is allowed to create a new message in a channel or throws an exception if not
+     *
+     * @param channel the channel the message should be created
+     * @param user    the user that wants to create the message
+     */
+    public void isAllowedToCreateNewAnswerPostInChannel(@NotNull Channel channel, @NotNull User user) {
+        var isArchivedChannel = !Objects.isNull(channel.getIsArchived()) && channel.getIsArchived();
+        var userToCheck = getUserIfNecessary(user);
+        var isChannelMember = isMember(channel.getId(), userToCheck.getId());
+        if (isArchivedChannel) {
+            throw new AccessForbiddenException("You are not allowed to create a new answer post in an archived channel.");
+        }
+        if (!isChannelMember) {
+            throw new AccessForbiddenException("User is not a member of the channel");
+        }
+    }
+
+    /**
+     * Checks if a user is allowed to create a new answer message in a channel or throws an exception if not
+     *
+     * @param channel the channel the answer message should be created
+     * @param user    the user that wants to create answer the message
+     */
+    public void isAllowedToCreateNewPostInChannel(@NotNull Channel channel, @NotNull User user) {
+        var isAnnouncementChannel = !Objects.isNull(channel.getIsAnnouncementChannel()) && channel.getIsAnnouncementChannel();
+        var isArchivedChannel = !Objects.isNull(channel.getIsArchived()) && channel.getIsArchived();
+        var userToCheck = getUserIfNecessary(user);
+
+        if (isArchivedChannel) {
```

Comment:
Same as above.

Reply:
Done in 7321b8ac348e12dee95c35679b2474892782d703